File: web/src/dtos/OrderItem.ts
import Common from "../utility/Common";
import { PagedReq } from "./PagedReq";
import { ProductRes } from "./Product";
import { UnitRes } from "./Unit";

export interface OrderItemRes {
  orderItemId?: number;
  orderId?: number;
  productId?: number;
  product?: ProductRes;
  quantity?: number;
  shippedQuantity?: number;
  rate?: number;
  basePrice?: number;
  unitId?: number;
  unit?: UnitRes;
}

export class OrderItemReqEdit {
  orderId?: number = 0;
  productId?: number = 0;
  quantity?: number = 1;
  rate?: number = 1;
  basePrice?: number = 1;
  unitId?: number = 0;
}

export class OrderItemReqSearch extends PagedReq {
  orderId?: number;
  unshippedItems?: boolean;

  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy,
      searchText = "",
    }: PagedReq,
    { orderId = 0, unshippedItems = false, }
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
    this.orderId = orderId;
    this.unshippedItems = unshippedItems;
  }
}

File: data/Dtos/OrderItem.cs
using data.Entities;
using Microsoft.EntityFrameworkCore.Metadata.Internal;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using data.Utility.Paging;

namespace data.Dtos
{
    public class OrderItemRes
    {
        public int OrderItemId { get; set; }
        public int OrderId { get; set; }
        public Order? Order { get; set; }
        public int ProductId { get; set; }
        public Product? Product { get; set; }
        public decimal Quantity { get; set; }
        public decimal ShippedQuantity { get; set; }
        public decimal Rate { get; set; }

        public decimal BasePrice { get; set; }

        public int? UnitId { get; set; }
        public Unit? Unit { get; set; }
    }

    public class OrderItemReqSearch : PagedRequestDto
    {
        public int OrderId { get; set; }
        public bool UnshippedItems { get; set; } = false;
    }

    public class OrderItemReqEdit
    {
        [Required]
        public int OrderId { get; set; }
        [Required]
        public int ProductId { get; set; }
        public decimal Quantity { get; set; }
        public decimal Rate { get; set; }
        public decimal BasePrice { get; set; }
        public int? UnitId { get; set; }
        
    }
}
